hil/cli.py

```diff
@@ -288,570 +190,352 @@ def serve_networks():
-        if len(args) == 4:
-            switchinfo = {"type": switch_api + subtype, "hostname": args[0],
-                          "username": args[1], "password": args[2],
-                          "interface_type": args[3]}
-        else:
-            sys.stderr.write('ERROR: subtype ' + subtype +
-                             ' requires exactly 4 arguments\n'
-                             '<hostname> <username> <password> '
-                             '<interface_type>\n'
-                             'NOTE: interface_type refers '
-                             'to the speed of the switchports\n '
-                             'ex. TenGigabitEthernet, FortyGigabitEthernet, '
-                             'etc.\n')
-            return
-    else:
-        sys.stderr.write('ERROR: Invalid subtype supplied\n')
-        return
-    url = object_url('switch', switch)
-    do_put(url, data=switchinfo)
-
-
-@cmd
-def switch_delete(switch):
-    """Delete a <switch> """
-    C.switch.delete(switch)
-
-
-@cmd
-def list_switches():
-    """List all switches"""
-    q = C.switch.list()
-    sys.stdout.write('%s switches :    ' % len(q) + " ".join(q) + '\n')
-
-
-@cmd
-def port_register(switch, port):
-    """Register a <port> with <switch> """
-    C.port.register(switch, port)
-
+###########################
+# I am doing the dash thing with the console commands. Let me know what's
+# preffered, should I go the nic commands route, or the network commands route.
 
-@cmd
-def port_delete(switch, port):
-    """Delete a <port> from a <switch>"""
-    C.port.delete(switch, port)
+@node.command(name='show-console', help='Show console')
+@click.argument('node')
+def node_show_console(node):
+    """Display console log for <node>"""
+    # Should be simple to implement in the client library for now.
+    C.node.show_console(node)
 
 
-@cmd
-def port_connect_nic(switch, port, node, nic):
-    """Connect a <port> on a <switch> to a <nic> on a <node>"""
-    C.port.connect_nic(switch, port, node, nic)
+@node.command(name='start-console', help='Start console')
+@click.argument('node')
+def node_start_console(node):
+    """Start logging console output from <node>"""
+    C.node.start_console(node)
 
 
-@cmd
-def port_detach_nic(switch, port):
-    """Detach a <port> on a <switch> from whatever's connected to it"""
-    C.port.detach_nic(switch, port)
+@node.command(name='stop-console', help='Stop console')
+@click.argument('node')
+def node_stop_console(node):
+    """Stop logging console output from <node> and delete the log"""
+    C.node.stop_console(node)
 
 
-@cmd
-def port_revert(switch, port):
-    """Detach a <port> on a <switch> from all attached networks."""
-    print C.port.port_revert(switch, port)
+####################################
+### PROJECT SUBCOMMANDS GO HERE! ###
+####################################
 
+###########################
+# I am invoking the project list and project list <something> commands here
```

Comment:
I kinda don't like this approach. I'd prefer to keep the groups and the commands separate. It feelds weird to have list the networks of a given project to be a sub-command of listing all the projects -- they don't feel related enough.

Reply:
this is what I think we can do:

`hil project list`  will list all projects
then we have the `hil project network` group
`hil project network list` will list all project networks
`hil project network grant-access` will grant project access to network
`hil project network revoke-access` will do the opposite

we can use the words add/remove in place of grant-access/revoke-access respectively.

Now while we are thinking about this. The way I have implemented connecting a project to a node is like this:
There's a group for project_connect, and then there's only thing that can connect to it i.e. nodes. so it reads like `hil project connect node`. Same thing for detach. This is inconsistent with what I am proposing. So we can change that to:

`hil project node list`
`hil project node add`
`hil project node remove`
and our network commands will read like
`hil project network list`
`hil project network add` (to grant access)
`hil project network delete` (to revoke-access)
`hil network attachments list` (note: hil network attachments then becomes a group).

Reply:
I don't have a strong opinion on this. I think this would work well. But I do believe that once we have made a decision, we should write something like an HIL community CLI guideline. I don't mind helping write or we could have it along with this PR 👍 @naved001 @zenhack @jeremy @Izhmash

Reply:
Sounds like a plan.